### PR TITLE
libcaca: update source and add x11 driver.

### PR DESCRIPTION
**Pull request by alxf**
Download source from official github repository tags and add libX11-devel as build depend for the x11 driver.

### CODE REVIEW

**Comment by Gottox** [comment]
Is the X11 driver useful for anyone? - I'm not sure if it just introduces an unused dependency. What do you guys think?

**Comment by Vaelatern** [comment]
@alxf For what reason do you think `libX11-devel` provides a useful component for this package?

**Comment by Gottox** [comment]
It's useful if you want performant caca rendering on X as most terminals are too slow. But I see no reason why we should have caca if there's graphical output anyway :)

**Comment by ebfe** [comment]
Make it build option, default to off?

**Comment by alxf** [comment]
Hi,

It was for performance reason, the rendering under the X11 driver is better. But indeed, it's not a real issue, if you mind to add this dep.

File: srcpkgs/libcaca/template [hunk 1]
```diff
@@ -6,12 +6,16 @@ build_style=gnu-configure
...
-makedepends="libX11-devel ncurses-devel"
+makedepends="ncurses-devel $(vopt_if x11 libX11-devel)"
 homepage="http://caca.zoy.org/wiki/libcaca"
 license="WTFPL"
 distfiles="https://github.com/cacalabs/libcaca/archive/v${version}.tar.gz"
 checksum=7ed29a00cc7f017424d8b2994f001f137ed5bc4441987b711d78c6734fdf3493
 
+# package build options
+build_options="x11"
+desc_option_x11="Enable x11 driver"
```
Comment (Vaelatern) [inline srcpkgs/libcaca/template]: `Enable X11 driver`

Comment (Duncaen) [inline srcpkgs/libcaca/template]: https://github.com/voidlinux/void-packages/blob/master/common/options.description
`desc_option_x11="Enable support for X11"`

Comment (Gottox) [inline srcpkgs/libcaca/template]: This line can be removed, as @Duncaen stated, there's a global descrition defined.

**Comment by alxf** [comment]
@Gottox 
That why i wanted to update the original source url, i'm not sure how you prefer to handle this diff, i can see with the project owner to make a new tag, and at this point only update source url with the real new release ?

**Comment by Vaelatern** [comment]
Since we don't have epochs yet I don't think we have a clean way to deal with a new release without a new number. Upstream making a new tag would be best.

**Pull request merged**